File: datasets/rs-keep-code_davinci_001_temp_0.2/HumanEval_2_truncate_number.rs
/// Given a positive floating point number, it can be decomposed into
/// and integer part (largest integer smaller than given number) and decimals
/// (leftover part always smaller than 1).
/// Return the decimal part of the number.
/// >>> truncate_number(3.5)
/// 0.5
fn truncate_number(number: f64) -> f64 {
    let mut number = number;
    let mut decimals = 0.0;
    while number > 1.0 {
        number /= 10.0;
        decimals += 1.0;
    }
    decimals

}

fn main() {
    let candidate = truncate_number;
    assert_eq!(candidate(3.5), 0.5);
    assert_eq!(candidate(1.25), 0.25);
    assert_eq!(candidate(123.0), 0.0);
}

File: datasets/lua-keep-incoder_6b_temp_0.2-0/HumanEval_2_truncate_number.lua
-- Given a positive floating point number, it can be decomposed into
-- and integer part (largest integer smaller than given number) and decimals
-- (leftover part always smaller than 1).
-- Return the decimal part of the number.
-- >>> truncate_number(3.5)
-- 0.5
local function truncate_number(number)
  if number < 1 then
    return number
  end
  local factor = 1
  local decimals = 0
  while number >= factor do
    factor = factor * 10
    decimals = decimals + 1
  end
  local integer_part = math.floor(number)
  local fractional_part = number - integer_part
  return integer_part + fractional_part / factor
end

-- Given a positive floating point number, it can be decomposed into
-- and integer part (largest integer smaller than given number) and decimals
-- (leftover part always smaller than 1).
-- Return the integer part of the number.
-- >>> truncate_number(3.5)
-- 3

lu = require('luaunit')

function test_humaneval()
local candidate = truncate_number
    lu.assertEquals(candidate(3.5), 0.5)
    lu.assertEquals(candidate(1.25), 0.25)
    lu.assertEquals(candidate(123.0), 0.0)
end

os.exit(lu.LuaUnit.run())